go/vt/vtgate/planbuilder/physical/subquery_planning.go

```diff
@@ -127,9 +130,59 @@ func mergeSubQueryOp(ctx *plancontext.PlanningContext, outer *Route, inner *Rout
 	if err != nil {
 		return nil, err
 	}
+
+	if subQueryWasPredicate {
+		// Copy Vindex predicates from the inner route to the upper route (probably not right)
```

Comment:
What do we mean by `probably not right`?

Reply:
I think that comment is outdated. I'll check back with @systay, but when I initially implemented this I was not sure if just copying the predicates over was the correct way of doing this.